src/main/xml-parsers.js

```diff
@@ -86,15 +77,19 @@ export function parseListMultipart(xml) {
     isTruncated: false
   }
   var xmlobj =  parseXml(xml)
-  if (xmlobj.IsTruncated && xmlobj.IsTruncated[0] === 'true') result.isTruncated = true
-  if (xmlobj.NextKeyMarker) result.nextKeyMarker =  xmlobj.NextKeyMarker[0]
+  if (!xmlobj.ListMultipartUploadsResult) {
+    throw new errors.InvalidXMLError('Missing tag: "ListMultipartUploadsResult"')
+  }
+  xmlobj = xmlobj.ListMultipartUploadsResult
+  if (xmlobj.IsTruncated && xmlobj.IsTruncated === 'true') result.isTruncated = true
+  if (xmlobj.NextKeyMarker) result.nextKeyMarker =  xmlobj.NextKeyMarker
   if (xmlobj.NextUploadIdMarker) result.nextUploadIdMarker = xmlobj.NextUploadIdMarker[0]
   if (xmlobj.CommonPrefixes) xmlobj.CommonPrefixes.forEach(prefix => {
     result.prefixes.push({prefix: prefix[0]})
   })
-  if (xmlobj.Upload) xmlobj.Upload.forEach(upload => {
+  if (xmlobj.Upload) (upload => {
```

Comment:
Yes, it is intentional.

With the new xml parser package, the returned `xmlobj.Upload` entity became a single element of type `any`, instead of being an array of type `any[]` as it was before with the older xml parser package.

Because of this difference, having the iteration causes the following expected error:
```
Uncaught TypeError: xmlobj.Upload.forEach is not a function
```

So, the iteration is needed  to be removed.

Reply:
@ebozduman as per the S3 spec `Upload` is `Array of MultipartUpload data types` https://docs.amazonaws.cn/en_us/AmazonS3/latest/API/API_ListMultipartUploads.html#API_ListMultipartUploads_ResponseSyntax

With the iteration removed, the result will not have all the uploads present.

Reply:
Then, this is a problem in the new XML parser package we are trying to update to.
Here is what the old and new XML parsers return:

Old:
```
xmlobj.Upload = [
  {
    Key: [ 'datafile-65-MB' ],
    UploadId: [ '9bcee11f-a707-4353-875a-147ca3dc472b' ],
    Initiator: [ [Object] ],
    Owner: [ [Object] ],
    StorageClass: [ '' ],
    Initiated: [ '2020-09-14T08:20:52.677Z' ]
  }
]
```

New:
```
xmlobj.Upload = {
  Key: 'datafile-65-MB',
  UploadId: '9f54503e-c712-48ea-a909-ca8bb5604f34',
  Initiator: { ID: '', DisplayName: '' },
  Owner: { ID: '', DisplayName: '' },
  StorageClass: '',
  Initiated: '2020-09-14T08:39:27.776Z'
}
```
Please notice `xmlobj.Upload` elements were also array before and they are single items in the new package.

Reply:
@ebozduman we could set the `arrayMode` with the value true so that the Upload is an array always. But this might need changes in other places as well.